### PR TITLE
Use command-line & waterfall for binaryen-shell

### PR DESCRIPTION
**Pull request by jfbastien**
This will allow me to run binaryen-shell tests on the waterfall.

### CODE REVIEW

**Comment by kripken** [comment]
Why did the known failures change? Waterfall update?

File: check.py [hunk 1]
```diff
@@ -195,21 +195,14 @@ def fail_if_not_contained(actual, expected):
           raise Exception('wasm interpreter error: ' + err) # failed to pretty-print
         raise Exception('wasm interpreter error')
 
-print '\n[ checking binaryen-shell... ]\n'
```
Comment (kripken) [inline check.py]: why did this go away? it checks we print `--help` with useful info.

**Comment by kripken** [comment]
Otherwise looks very nice!

**Comment by jfbastien** [comment]
I was about to ask you why more tests pass. Had you triaged any of them?

File: check.py (hunk 1 above)
Comment (jfbastien) [inline check.py]: All the programs are using the same command-line support, so they either all work on none of them work ;-)

I can add a separate command-line only test, but that seemed like overkill.

**Comment by kripken** [comment]
No, I just opened an issue about triaging them :)

If no one looked into them yet, I guess we can leave figuring out the new passes for proper triage anyhow.

**Comment by jfbastien** [comment]
Yeah, I'm going to run the same tests on other VMs so my thinking is that we'll find any issues that are hiding anyways.

File: check.py (hunk 1 above)
Comment (kripken) [inline check.py]: But this tested the contents of what the user can read when `--help` is requested. We don't check that anywhere else, do we?

Comment (jfbastien) [inline check.py]: Correct, it's not tested elsewhere. It just seemed arbitrary to test it just for the shell, it seems better to test all the bin/ content or none of it. I can add such a test if you want.

Comment (kripken) [inline check.py]: Ok, the reason this was tested is that the shell is the most user-facing tool here - I want stuff to break if we make that less useful for people :)

**Pull request merged**